probe-rs/src/architecture/arm/memory/adi_v5_memory_interface.rs

```diff
@@ -321,9 +321,9 @@ where
             Ok(())
         } else {
             self.interface
-                .read_ap_register_repeated(self.memory_ap, DRW { data: 0 }, values)
+                .read_ap_register_repeated(self.memory_ap.clone(), DRW { data: 0 }, values)
```

Comment:
Looking at this, the clone and pass-by-value looks like an inconsistency. In `read_ap_register` the address is taken by reference. Can we please have that here (and in the writes), too? I'd prefer not introducing infinite amounts of cloning here, even if this code will only meet with v1 addresses. I feel like this would establish a less than ideal precedence for when v2 is actually handled.

Reply:
Although the other method where expecting a `impl into<PORT>` (± `Clone`) it wasn’t actually required in any of them, so I did remove the unecessary `clone()`.